traffic_ops/testing/api/v14/serverservercapability_test.go

```diff
@@ -0,0 +1,167 @@
+		ServerID: ssc.ServerID,
+	}
+	_, _, err = TOSession.CreateServerServerCapability(sscNilCapability)
+	if err == nil {
+		t.Errorf("expected to receive error when assigning a server capability to a server without a server capability\n")
+	}
+
+	// Attempt to assign an server capability with invalid server capability
+	sscInvalidCapability := tc.ServerServerCapabilityNullable{
+		ServerID:         ssc.ServerID,
+		ServerCapability: util.StrPtr("bogus"),
+	}
+	_, _, err = TOSession.CreateServerServerCapability(sscInvalidCapability)
+	if err == nil {
+		t.Errorf("expected to receive error when assigning a non existent server capability to a server\n")
+	}
+
+	// Attempt to assign an server capability with invalid server capability
+	sscInvalidID := tc.ServerServerCapabilityNullable{
+		ServerID:         util.IntPtr(-1),
+		ServerCapability: ssc.ServerCapability,
+	}
+	_, _, err = TOSession.CreateServerServerCapability(sscInvalidID)
+	if err == nil {
+		t.Errorf("expected to receive error when assigning a server capability to a non existent server ID\n")
+	}
+}
+
+func GetTestServerServerCapabilities(t *testing.T) {
+	// Get All Server Capabilities
+	sscs, _, err := TOSession.GetServerServerCapabilities(nil, nil, nil)
+	if err != nil {
+		t.Fatalf("cannot GET server capabilities assigned to servers: %v\n", err)
+	}
+	if sscs == nil {
+		t.Fatalf("returned server capabilities assigned to servers was nil\n")
+	}
+	if len(sscs) != len(testData.ServerServerCapabilities) {
+		t.Errorf("expect %v server capabilities assigned to servers received %v \n", len(testData.ServerServerCapabilities), len(sscs))
+	}
+
+	for _, ssc := range sscs {
+
+		checkResp := func(t *testing.T, sscs []tc.ServerServerCapabilityNullable) {
+			if sscs == nil {
+				t.Fatalf("returned server capabilities assigned to servers was nil\n")
+			}
+			if len(sscs) != 1 {
+				t.Errorf("expect 1 server capabilities assigned to server received %v \n", len(sscs))
+			}
+		}
+		// Get assigned Server Capabilities by server id
+		sscs, _, err := TOSession.GetServerServerCapabilities(ssc.ServerID, nil, nil)
+		if err != nil {
+			t.Fatalf("cannot GET server capabilities assigned to servers by server ID %v: %v\n", *ssc.ServerID, err)
+		}
+		checkResp(t, sscs)
+		// Get assigned Server Capabilities by host name
+		sscs, _, err = TOSession.GetServerServerCapabilities(nil, ssc.Server, nil)
+		if err != nil {
+			t.Fatalf("cannot GET server capabilities assigned to servers by server host name %v: %v\n", *ssc.Server, err)
+		}
+		checkResp(t, sscs)
+
+		// Get assigned Server Capabilities by server capability
+		sscs, _, err = TOSession.GetServerServerCapabilities(nil, nil, ssc.ServerCapability)
+		if err != nil {
+			t.Fatalf("cannot GET server capabilities assigned to servers by server capability %v: %v\n", *ssc.ServerCapability, err)
+		}
+		checkResp(t, sscs)
+	}
+}
+
+func DeleteTestServerServerCapabilities(t *testing.T) {
+	time.Sleep(10 * time.Minute)
```

Comment:
So this is why my TO API test run is hanging!

Reply:
hahah shoot! was in for debugging and forgot to remove :)